### PR TITLE
[#6] [Android] [UI] As a user, I can sign in with email and password

### LINKED ISSUE #6: [Android] [UI] As a user, I can sign in with email and password
## Why

Users must be authenticated before taking or viewing any surveys. Our surveys have access restrictions. Some surveys are targeted to a specific group of users only for a more accurate survey result.

The mobile app should allow users to authenticate with their email and password.

## Acceptance Criteria

- Display the Nimble logo.
- Display the blurred background.
- Display the Email text input.
- Set the placeholder text to Email.
- Display the Password text input.
- Set the placeholder text to Password.
- Mask the password input with the native-OS feature.
- Display the Login button.
- **DO NOT** display the Forgot Password button yet. We will work on this on another separate epic.

## Design

Image

Link: https://www.figma.com/file/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=31%3A195

### PR DESCRIPTION
**Pull request by suho**
Resolves #6

## What happened 👀

- Added login screen with animation
- Added UI test for the login screen

## Insight 📝

- I created `PrimaryButton` and `PrimaryTextField` composable to support reuse later
- I used different ways to apply animation for the login screen
    - `AnimatedVisibility` to support showing and hiding the UI components
    - `animateFloatAsState`, `animateOffsetAsState` to implement animation based on value changes

### CODE REVIEW

**Comment by luongvo** [comment]
@suho Should we have some more detail at `Insight` section about what you do with animation?

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/login/LoginScreen.kt [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        blurState = 20f
+    }
+    Image(
+        painter = painterResource(id = R.drawable.bg_splash),
+        contentDescription = null,
+        contentScale = ContentScale.FillWidth,
+        modifier = modifier
+            .blur(radius = animateBlurState.dp)
+    )
+}
```
Comment (luongvo) [inline android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/login/LoginScreen.kt:107]: I don't see the blurState working 🤔 

- Pixel 2XL - Android 11



https://user-images.githubusercontent.com/16315358/192937871-61763358-e56e-4533-8a38-c3dbcdb2bb0e.mp4

Comment (luongvo) [reply android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/login/LoginScreen.kt:107]: The `modifier.blur` does not seem to work on Android 11 and lower https://www.reddit.com/r/mAndroidDev/comments/v62ukr/blur_in_jetpack_compose_supported_only_in_android/. Flutter makes life easier 😮‍💨 

- Pixel 6 - Android 13 ✅ 


https://user-images.githubusercontent.com/16315358/192938382-c855c54a-2404-4818-bd22-42a056e5f794.mp4

File: android/src/androidTest/java/co/nimblehq/ic/kmm/suv/android/ui/screens/login/LoginScreenTest.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    @Test
+    fun when_enter_the_login_screen__it_shows_email_password_and_login_button() {
+        composeRule.apply {
+            mainClock.advanceTimeBy(4000) // After animation
+            onNodeWithText(activity.getString(R.string.login_email)).assertIsDisplayed()
+            onNodeWithText(activity.getString(R.string.login_password)).assertIsDisplayed()
+            onNodeWithText(activity.getString(R.string.login_button)).assertIsDisplayed()
+        }
```
Comment (luongvo) [inline android/src/androidTest/java/co/nimblehq/ic/kmm/suv/android/ui/screens/login/LoginScreenTest.kt:34]: 👍

**Review by luongvo — COMMENTED** [review]
rest lgtm

**Comment by suho** [comment]
@luongvo I fixed all your comments here: https://github.com/suho/kmm-ic/pull/84/commits/f1be7bfe709784c17ab5ffed1489e948cff3da36

**Review by luongvo — APPROVED** [review]
lgtm now

**Review by vnntsu — APPROVED** [review]
LGTM! 💪

**Pull request merged**